tests/test_appname_cmd.py

```diff
@@ -1,2 +1,12 @@
-def test_app_name(cmd):
-    assert cmd.get_app_name() == "Boilerplate"
+from application_client.boilerplate_cmd import BoilerplateCommand
+from application_client.boilerplate_response_unpacker import unpack_get_app_name_response
+
+
+# In this test we check that the GET_APP_NAME replies the application name
+def test_app_name(backend, firmware):
+    # Use the app interface instead of raw interface
+    client = BoilerplateCommand(backend)
+    # Send the GET_APP_NAME instruction to the app
+    response = client.get_app_name()
+    # Assert that we have received the correct appname
+    assert unpack_get_app_name_response(response.data) == "Boilerplate"
```

Comment:
I disagree.
For the asynchronous call, there is no way for the client to automatically unpack the response.
So IMO the generic way would to have a unpack method for each APDU response.

But anyway, I think that in python of the motto is `Simple is better than complex`, so we should keep to a simple class and not add too much abstraction layers.

Reply:
Indeed `Simple is better than complex`, but also `Complex is better than complicated`. I feel like a code like

```python
    with client.get_public_key_with_confirmation(path=path):
        if backend.firmware.device == "nanos":
            nav_ins = create_simple_nav_instructions(5)
        elif backend.firmware.device.startswith("nano"):
            nav_ins = create_simple_nav_instructions(3)
        navigator.navigate_and_compare(ROOT_SCREENSHOT_PATH, test_name, nav_ins)
    response = client.get_async_response().data
    _, public_key, _, chain_code = unpack_get_public_key_response(response)

    ref_public_key, ref_chain_code = calculate_public_key_and_chaincode(CurveChoice.Secp256k1, path=path)

    assert public_key.hex() == ref_public_key
    assert chain_code.hex() == ref_chain_code
```

... is **very** complicated. And is a lie too, because clearly `client.get_public_key_with_confirmation` does not return the public key, but rather some mystic environment which allows me to extract the key. As a **client** user, I really don't care about all this shenanigans - and I will certainly not spend much time figuring out how it works.

The code I wrote should be

```python
public_key, chain_code =  client.get_public_key_with_confirmation(path=path)

assert public_key.hex() == ref_public_key
assert chain_code.hex() == ref_chain_code
```
... and that's it.

Again, it's my PoV of what a client should be. Just renaming all `client` variable with something clearly stating it is nothing more than a mere helper would do for me, and I can cope that we change nothing at all.

Reply:
As an example, [this is what the exchange tests look like](https://github.com/LedgerHQ/app-exchange/blob/develop/test/python/test_fund_flow_ethereum.py).

Indeed the client is a bit complex, but the tests are straightforwards, and if anybody wants to use the client to tinker with the app, I sincerely thinks it is much easier.

Reply:
The real diff here is that the client is handling the device interactions (for example right clicks) which are passed as parameter. I'm afraid it's not going to be so simple if we want to:
- handle more click scenarios (like refusal, or going back in the flow, or interacting with the settings)
- handle Ledger Stax interactions
- handle inter_interactions screenshot comparisons.

All of this which can be easily done on the test side with either `navigate_and_compare_until_text`, or `navigate_until_text` or `navigate_and_compare` depending on what is wanted by the test.

This is at the cost that the client can only expose a `send` function and a `parse_response` function, but I don't find it so complex as:
- there are not dozens of different methods
- they don't takes dozen of parameters
- their names can be carefully chosen so that they explain what method is `async` (hence should be called with `with` keyword) and what method is a response parser for which response.